packages/apiserver/src/resources/materials/material.resolver.ts

```diff
@@ -105,17 +105,8 @@ export class MaterialResolver {
   async removeMaterial(
     @Args('input') removeMaterialInput: RemoveMaterialInput
   ) {
-    try {
-      const identifier = removeMaterialInput.identifier;
-      const material = await Material.findOne({
-        where: { identifier },
-      });
-      await Status.delete({ material_id: material.id });
-      await Material.remove({ ...material } as Material);
-      return material;
-    } catch (e) {
-      throw new BadRequestException();
-    }
+    const identifier = removeMaterialInput.identifier;
+    return await this.materialService.deleteMaterial(identifier);
```

Comment:
better wrap it with `try/catch`

Reply:
I don't think rethrowing error here is needed.
Do I miss something?

Reply:
Do we handle such errors somewhere else? Like interception filter or so?

Reply:
No, we don't have custom filter. GraphQL will handle it automatically and will add errors in the response. Since the error thrown (potentially) doesn't extend HttpException, it will be INTERNAL_SERVER_ERROR. I think this is desired behavior in this case